### ISSUE TITLE
Best way to fit multi-task model with independent tasks?

### ISSUE THREAD

**Issue by eytan** [cornellius-gp/gpytorch#318]
I am doing some comparison across multi-task models and would like to compare against a model that models each task independently.  Is there an easy way to specify such a MT kernel?  Ideally it would not require greater time/space complexity than fitting each task separately.

**Comment by jacobrgardner** [comment]
My recommendation would be to just take advantage of the batch mode functionality we added that lets you train multiple GPs simultaneously.

I suppose you could also handle this as multitask by initializing the task covariance matrix to identity and setting `requires_grad` to True, but it sounds like your problem is exactly what we implemented batch mode for :-)

**Comment by jacobrgardner** [comment]
See https://github.com/cornellius-gp/gpytorch/blob/master/examples/01_Simple_GP_Regression/Simple_Batch_Mode_GP_Regression.ipynb

**Comment by eytan** [comment]
Initializing the task covariance to the identity and turning off autograd sounds like an elegant solution! Is there any reason to prefer doing things with batch mode?  One advantage is that it would make it easier to use ARD with different lengscales in batch mode.

**Comment by jacobrgardner** [comment]
Tl;dr: I would anticipate batch mode being faster. 

Long version: the way multitask is implemented is by forming a `KroneckerProductLazyVariable` which would have the same running time if the Identity matrix formed was actually a `DiagLazyVariable`, but since the multitask code expects the task covariance to be any matrix it doesn't make a `DiagLazyVariable` so may incur the cost of multitask GPs without actually doing anything multitask.

**Comment by Balandat** [comment]
@eytan, the question is also whether you want a single set of hyperparameters shared across the different tasks, which is what the MTGPs do. In standard batch mode you have fully independent GPs (unless you do share the hyperparameters across batches, which is also possible, see https://github.com/cornellius-gp/gpytorch/blob/master/test/examples/test_batch_gp_regression.py#L156).

**Comment by eytan** [comment]
I think batch mode makes the most sense here. I just want to make sure that it maintains the same interface as a MT-GP, as to more easily facilitate direct comparison across independent GP and Multi-task methods.

**Comment by gpleiss** [comment]
This should actually be pretty seamless now with a commit from #903 . Train a batch-mode GP (one task per independent batch). Then call `MultitaskMultivariateNormal.from_batch_mvn(likelihood(model(test_x)))` - and you should get a MMVN output.
